### DISCUSSION TITLE
Desktop web browsers can't play 4k videos, any solution?

### DISCUSSION THREAD

**Discussion by joe2xyz** [blakeblackshear/frigate#10540] [General]
Hi folks,

A few days ago I started using Frigate with Reolink RLC-811A and it works very well, thank you for the fantastic project!

The only problem I have is that both Firefox and Chromium (on Linux Desktop) are unable to play recorded clips regardless if I use an h264 or h265 camera stream - error: `Playback cannot continue. No available working or supported playlists.`

After some research I realized that this isn't an issue in Frigate, it's a known problem that web browsers can't play 3840x2160px video. I tested that simply by opening a 3840x2160px mp4 file in Firefox and it couldn't play it. VLC played it without any issues.

After more research I found that I can't downscale using Frigate "restream" but I don't want to do that. I wouldn't buy a 4k camera to record 2k videos...

Is there a simpler solution instead of downloading a video and opening it in VLC?

Thank you for the amazing project!

Kind regards,

Joe

**Comment by NickM-27** [comment]
if your browser can't play the video then there is not much that can be done. That said, many users record and view h.265 4K video without issues. 

Firefox does not support h.265 and linux will depend on the distro and gpu installation.

**Reply by NickM-27** [reply]
also just to note that reolink does not support recording 4k resolution on h.264, so not sure how that was tested

**Comment by joe2xyz** [comment]
> reolink does not support recording 4k resolution on h.264

Are you sure about that? I'm using `rtsp://admin:...@...:554/h264Preview_01_main` just fine.

I have the most current firmware so maybe that's the reason.

**Reply by NickM-27** [reply]
you can use whatever URL you want, on my h264 reolink doorbell I can use h265Preview_01_main and that does not mean it actually changes what is sent. If you check VLC stream info you will see it is h265 no matter if you put h264 or h265 in front of Preview_01_main (you can actually omit it altogether and just use Preview_01_main)

**Comment by DjeiiStylz** [comment]
Hi mate,
I use Edge to play H265 reolink on windows, Safari on mac, and frigate viewer on Android. 
It work well. Didnt tried on linux. Maybe u can find a package to play h265 on linux?

**Comment by joe2xyz** [comment]
> If you check VLC stream info

I had no idea! Thank you.

> I use Edge to play H265 reolink on windows

Thanks for the tip @DjeiiStylz ! I did some initial googling and didn't find a solution but I'll try again tomorrow.

Could you please try Firefox on Windows (I have no Windows to test it...) ?

**Comment by DjeiiStylz** [comment]
> Could you please try Firefox on Windows (I have no Windows to test it...) ?

Firefox don't work on H265 reolink camera. It start to play, but after few second it crashed with the following message : " Playback cannot continue. No available working or supported playlists." 
Same on chrome. 

Only edge works on windows. ( I used this link to figure it out)

**Reply by Pflaumenaugust** [reply]
I use a Reolink 811 with Frigate. On Windows, I can play the recordings with Chrome in H265 with no Problems.

**Reply by DjeiiStylz** [reply]
Just tried to enable it on Firefox on Windows. Everything work too. 
You need to enable "media.wmf.hevc.enabled" in about:config. (0->1) and restart Firefox. 
Actually working on Reolink cameras H265.
